File: src/builtins/exit_command.nim
import builtins/builtin_command
import core/session_state

type
  ExitCommand* = ref object of BuiltInCommand

proc newExitCommand*(): ExitCommand =
  ExitCommand(name: "exit")

method run*(self: ExitCommand, args: seq[string], state: SessionState) =
  state.cont = false

File: src/builtins/exit_command.rs
use crate::builtins::builtin_command::BuiltInCommand;
use crate::core::session_state::SessionState;

pub struct ExitCommand;

impl BuiltInCommand for ExitCommand {
    fn name(&self) -> &str {
        "exit"
    }

    fn run(&self, _args: &[String], state: &mut SessionState) {
        state.cont = false;
    }
}
